### ISSUE TITLE
[sklearn#34437] The PyPI wheel doesn't bundle the right `libgomp` on Linux

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#324]
**Mirrored from [scikit-learn#34437](https://github.com/scikit-learn/scikit-learn/issues/34437)**

Original title: The PyPI wheel doesn't bundle the right `libgomp` on Linux
Original labels: Bug, Needs Investigation
Created: 2026-07-06T12:31:09Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

### Describe the bug and give evidence about its user-facing impact

The PyPI wheel bundles `libgomp.so.1.0.0-8.5.0-28.el8_10.alma.1.x86_64.debug` which is probably not what we want.

Despite scikit-learn [building in manylinux_2_28](https://github.com/scikit-learn/scikit-learn/blob/b2958e8ac0da5f56fe82677c2a67ed960f8d3a89/.github/workflows/wheels.yml#L100), whose [docs say](https://github.com/pypa/manylinux#manylinux_2_28-almalinux-8-based) the toolchain is GCC 14, the bundled OpenMP runtime is effectively AlmaLinux 8 / GCC 8.5-era libgomp.

Codex suggests to add this repair command around [here](https://github.com/scikit-learn/scikit-learn/blob/b2958e8ac0da5f56fe82677c2a67ed960f8d3a89/.github/workflows/wheels.yml#L194) to the linux build:

```yaml
          CIBW_REPAIR_WHEEL_COMMAND_LINUX: >
            LD_LIBRARY_PATH=/opt/rh/gcc-toolset-14/root/usr/lib64:$LD_LIBRARY_PATH
            auditwheel repair -w {dest_dir} {wheel}
```

`auditwheel repair` decides which concrete shared object to copy into `scikit_learn.libs` (what is bundled in the PyPi wheel). Without help, it resolves the base AlmaLinux 8 runtime. With `LD_LIBRARY_PATH` pointing to gcc-toolset first, it should resolve to the newer gcc-toolset libgomp.

### Steps/Code to Reproduce

Run from a PiPy sklearn on Linux:

```Python
import threadpoolctl

for info in threadpoolctl.threadpool_info():
    if info.get("prefix") == "libgomp":
        print(info["filepath"])
        break
```

Then in bash:

```bash
LIBGOMP="<what was printed by the python snippet>
strings "$LIBGOMP" | grep -E "libgomp.so.1.0.0-|annobin gcc|running gcc"
```

You'll see something like:
```
annobin gcc 8.5.0 20210514
running gcc 8.5.0 20210514
libgomp.so.1.0.0-8.5.0-28.el8_10.alma.1.x86_64.debug
```

### Expected Results

N/A

### Actual Results

N/A

### Versions

```shell
System:
    python: 3.12.13 | packaged by conda-forge | (main, Mar  5 2026, 16:50:00) [GCC 14.3.0]
executable: /home/arthur/probabl/scikit-learn-benchmarks/.pixi/envs/sklearn/bin/python
   machine: Linux-6.17.0-35-generic-x86_64-with-glibc2.39

Python dependencies:
      sklearn: 1.9.0
          pip: None
   setuptools: None
        numpy: 2.4.6
        scipy: 1.17.1
       Cython: None
       pandas: 2.3.3
   matplotlib: 3.11.0
       joblib: 1.5.3
threadpoolctl: 3.6.0
     narwhals: 2.23.0

Built with OpenMP: True

threadpoolctl info:
       user_api: blas
   internal_api: openblas
    num_threads: 16
         prefix: libopenblas
       filepath: /home/arthur/probabl/scikit-learn-benc

...(truncated)

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/99 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
